### PR TITLE
 Allow API authentication using the `auth` query parameter containing an auth. hash.

### PR DESCRIPTION
**Pull request by mnoorenberghe**
This implements #1827. Let's try to keep the discussion about the concept there and focus this pull request on the technical details of the implementation to avoid fragmenting the conversation.

The reason for moving the global `Debug` function was because it conflicted with a CakePHP method of the same name. I thought of a few different approaches to workaround the problem but I settled on moving it to a static method on Logger as it seemed cleanest.
> PHP Fatal error:  Cannot redeclare Debug() (previously declared in /usr/local/share/zoneminder/www/api/lib/Cake/basics.php:73) in /usr/local/share/zoneminder/www/includes/logger.php on line 508

### CODE REVIEW

**Comment by mnoorenberghe** [comment]
Gist with a shell script to test some combinations: https://gist.github.com/mnoorenberghe/f59156f0c2fe15467f4745f8204a3edd

File: web/ajax/stream.php [hunk 1]
```diff
@@ -21,23 +21,23 @@
 switch ( $_REQUEST['command'] )
 {
     case CMD_VARPLAY :
-        Debug( "Varplaying to ".$_REQUEST['rate'] );
+        Logger::Debug( "Varplaying to ".$_REQUEST['rate'] );
```
Comment (connortechnology) [inline web/ajax/stream.php:24]: Why put the Logger:: in front?  It's outside the scope of the PR.  I actually do like to fully specify function names, but this seems unnecessary.

Comment (mnoorenberghe) [reply web/ajax/stream.php:24]: This is explained in the description of the pull request

File: web/index.php [hunk 2]
```diff
@@ -175,8 +175,8 @@
 isset($request) || $request = NULL;
 isset($action) || $action = NULL;
 
-if ( ZM_ENABLE_CSRF_MAGIC && $action != 'login' ) {
-    Debug("Calling csrf_check with the following values: \$request = \"$request\", \$view = \"$view\", \$action = \"$action\"");
+if ( ZM_ENABLE_CSRF_MAGIC ) {
```
Comment (connortechnology) [inline web/index.php:178]: You took out the && $action != 'login' which is there for a reason.

Comment (mnoorenberghe) [reply web/index.php:178]: Right, I took it out since it was only a workaround for not having a proper way to authenticate to the API which this bug is resolving.

File: web/includes/functions.php [hunk 3]
```diff
@@ -27,6 +27,9 @@ function ob_get_clean() {
   }
 }
 
+require_once( 'logger.php' );
+require_once( 'database.php' );
```
Comment (connortechnology) [inline web/includes/functions.php:31]: Why did you add these?  by the time anything includes functions.php it is generally understood that these will have been included already, in index.php.  Did you find that they were necessary?

**Comment by connortechnology** [comment]
I am merging this to my storageareas branch to give it some testing.  

I get why we need the Logger::Debug change.. but I don't like it.  If we are going to make Debug into a static method, then I think Info and Warning and the rest should be too.  

I'm kinda wondering if we shouldn't do something like $Log->Debug() instead.  It's more what I am used to doing in my code.  Not saying it's better, but maybe it is.  Opinions?

File: web/index.php (hunk 2 above)
Comment (knight-of-ni) [reply web/index.php:178]: While I do agree our goal should be to remove $action != login, I am concerned about the timing. In order to minimize the pain, perhaps we should first document the old auth method is deprecated, and then remove it in a future version.
